### PR TITLE
Adding Keystone V3 settings into openrc-maas along with endpoint type selection

### PR DESCRIPTION
**Pull request by BjoernT**
Adding V3 settings into openrc-maas, similar to standard openrc.
Also adding endpoint type selection and a new setting for insecure API calls (OSA fix pending LP #1537117)
Also fixing nova_exc.Unauthorized error for nova to reauthenticate instead
of only expecting AttributeError

Closes-Bug: #763 
Closes-Bug: #765

### CODE REVIEW

File: maas/plugins/maas_common.py [hunk 1]
```diff
@@ -351,13 +378,22 @@ def get_service_catalog(auth_ref):
                         auth_ref.get('serviceCatalog'))
 
 
+def get_endpoint_type(auth_details):
```
Comment (sigmavirus24) [inline maas/plugins/maas_common.py]: Can we just do

```py
def get_endpoint_type(auth_details):
    endpoint_type = auth_details['OS_ENDPOINT_TYPE']
    if endpoint_type == 'publicURL':
        return 'public'
    if endpoint_type == 'adminURL':
        return 'admin'
    return 'internal'
```

Comment (BjoernT) [inline maas/plugins/maas_common.py]: sure

**Comment by sigmavirus24** [comment]
Other than my one comment I have no other review comments. Thanks @BjoernT

**Comment by prometheanfire** [comment]
Generally seems alright, would like to get someone who's done more maas work to review first though.

**Comment by d34dh0r53** [comment]
:+1:

**Pull request merged**